azure-iot-hub/azure/iot/hub/iothub_registry_manager.py

```diff
@@ -56,7 +56,7 @@ class IoTHubRegistryManager(object):
     based on top of the auto generated IotHub REST APIs
     """
 
-    def __init__(self, connection_string=None, host=None, auth=None):
+    def __init__(self, connection_string=None, host=None, token_credential=None):
```

Comment:
This change would break users who are using AAD auth and use the `IoTHubRegistryManager` constructor directly. I think this is fine because:

1. The AAD feature has not been out for very long, so it would only affect very early adopters of this feature
2. Users should not be using the constructor directly anyway, they should be using the `from_token_credential()` function. It's not even that easy to do this by mistake because it would require the user to manually pass in an `AzureIdentityCredentialAdapter` instance. 

Thoughts? @BertKleewein

Reply:
I agree.   Since these parameters have only just been added and hubs that support them aren't even fully online, it's safe to assume that nobody uses them.  I would add 2 things to the convesrsation:

1. This does not break existing customers who use the IoTHubRegistryManager constructor with a connection string since the name and position of this argument hasn't changed.

2. We should also document that customers shouldn't call this directly and should use a factory method instead.

Reply:
It seems like most of our samples call the constructor directly instead of using a factory method. Maybe we should fix that.

Reply:
Done